src/network/networkd-ndisc.c

```diff
@@ -839,6 +839,11 @@ static int ndisc_router_process_route(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return log_link_error_errno(link, r, "Failed to get gateway address from RA: %m");
 
+        if (in6_addr_compare_func(&link->ipv6ll_address, &gateway) == 0) {
+                log_link_warning(link, "Advertised route gateway is local to the link, ignoring route");
```

Comment:
I think logging in DEBUG level is sufficient. Please use `log_link_debug()`

Also, it is better to include the provided gateway address in the log message.